File: Framework/LanguageSupport/cs/MMICSharp/MMIStandard/PropertiesCreator.cs
// SPDX-License-Identifier: MIT
// The content of this file has been developed in the context of the MOSIM research project.
// Original author(s): Felix Gaisbauer

using System.Collections.Generic;

namespace MMICSharp.MMIStandard.Utils
{
    public class PropertiesCreator
    {
        public static Dictionary<string,string> Create(params string[] properties)
        {
            Dictionary<string, string> dict = new Dictionary<string, string>();

            for(int i=0; i< properties.Length; i += 2)
            {
                dict.Add(properties[i], properties[i + 1]);
            }

            return dict;
        }
    }
}

File: Framework/EngineSupport/Unreal/MOSIMPlugin/Source/MMMIScene/Private/PropertiesCreator.cpp
// SPDX-License-Identifier: MIT
// The content of this file has been developed in the context of the MOSIM research project.
// Original author(s): Stephan Adam

// Helper class for creating the properties of the MInstructions

#include "PropertiesCreator.h"

map<string, string> PropertiesCreator::Create( const vector<string>& properties )
{
    // check, if the properties vector is of even size
    if( properties.size() % 2 != 0 )
        throw runtime_error( "Properties vector has to be of even length!" );

    map<string, string> _return = map<string, string>();
    for( int i = 0; i < properties.size(); i += 2 )
    {
        _return.insert( pair<string, string>{properties[i], properties[i + 1]} );
    }
    return _return;
}